src/client/sandbox/node/window.js

```diff
@@ -95,6 +110,18 @@ export default class WindowSandbox extends SandboxBase {
             return false;
         };
 
+        window.setTimeout = function () {
+            var args = WindowSandbox._processTimerArgs(arguments);
```

Comment:
Do not leak arguments (see: https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#32-leaking-arguments). Copy them to array in-place instead. Add link to this article here